test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Formatter/InheritanceTests.cs

```diff
@@ -21,21 +16,22 @@
 using Microsoft.Test.E2E.AspNet.OData.Common.Instancing;
 using Microsoft.Test.E2E.AspNet.OData.Common.Models.Vehicle;
 using Xunit;
+using ODataPath = Microsoft.AspNet.OData.Routing.ODataPath;
 
 namespace Microsoft.Test.E2E.AspNet.OData.Formatter
 {
-    public class DeleteAllRoutingConvention : EntitySetRoutingConvention
+    public class DeleteAllRoutingConvention : TestEntitySetRoutingConvention
     {
-        public override string SelectAction(Microsoft.AspNet.OData.Routing.ODataPath odataPath, HttpControllerContext context,
-            ILookup<string, HttpActionDescriptor> actionMap)
+        /// <inheritdoc/>
+        protected override string SelectAction(string requestMethod, ODataPath odataPath, TestControllerContext controllerContext, IList<string> actionList)
         {
-            if (context.Request.Method == HttpMethod.Delete &&
-                odataPath.PathTemplate == "~/entityset")
+            if (requestMethod == "DELETE" &&
```

Comment:
Is there a const string defined somewhere for the DELETE operation?

Reply:
Not a good universal one sadly. AspNetCore offers an HttpMethod class with string constants but AspNet used the System.Web verison of HttpMethods which are static class instances backed by strings. Using raw strings is the most compatible version.

The product code get's around this with an abstracted version of HttpMethod, ODataRequestMethod which is an int-based enum, and the adapter classes do the appropriate conversion. While that is possible on the test side, it seems like overkill.